added documentation for new producers (#565)
nasa-gcn/gcn.nasa.gov: app/routes/docs/newproducers.mdx

into setting up new notices streams that are distributed to the user community
via [Kafka](faq#what-is-kafka). This process requires interactions
with the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
to enable these accounts and topics are setup on the GCN Kafka broker,
and we are happy to work together to help you define schema utilizing our
core JSON schema.

## Defining Kafka Topics and JSON Schema

The preferred format for formatting of the data sent via GCN Kafka is JSON.
We have designed a set of [core schema](https://github.com/nasa-gcn/gcn-schema/tree/master/core-schema)
which serves as the building blocks for your Notices.
You can also define instrument-specific quantities in your
schema, but we request that you utilize the core schema as much as possible.
Please add your schema to the [GCN repository](https://github.com/nasa-gcn/gcn-schema).

The GCN team is happy to iterate with the producers on their schema contents and format.
Your pipeline will generate JSON files following these schema and send to GCN
in the following process.

## Steps to Start Publishing Alerts

> 1. Make sure that all of the producer users (who will enable GCN notices in your pipelines)

Reviewer: who will enable?

   have signed in (at least once) to https://test.gcn.nasa.gov and https://gcn.nasa.gov.
2. Please send the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
   a list of email addresses of all the people who should have the

>    producer access to the MissionName.\* topics.

Reviewer: Italicize _mission_name_ so that it is clear that it is template text:

```suggestion
   producer access to the <code><i>mission_name</i>.*</code> topics.
```

3. Log out and log back in.

> 4. Go through the "Start Streaming GCN Notices" process.

Reviewer: Make this a link...

> 5. On Step 2 of "Start Streaming GCN Notices", choose the scope gcn.nasa.gov/kafka-MissionName-producer.

Reviewer: ```suggestion
5. On Step 2 of "Start Streaming GCN Notices", choose the `scope gcn.nasa.gov/kafka-mission_name-producer`.
```

> 6. Your producer code will look very similar to the client example code.

Reviewer: Add link...

>    Start from this and adjust as you see fit:

Reviewer: Explain where the client ID and client secret come from...

```sh
from gcn_kafka import Producer
# Connect as a producer.
# Warning: don't share the client secret with others.
producer = Producer(client_id='fill me in', client_secret='fill me in')

> topic = 'MissionName.example'

Reviewer: ```suggestion
topic = 'mission_name.example'
```

# anything starting with 'MissionName'
data = b'...'  # any bytes
producer.produce(topic, data)
```

## Mission page

Please create a new mission page by submitting a [pull request](https://github.com/nasa-gcn/gcn.nasa.gov/pulls)
or by emailing text to the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn).

## Announce New Notices Type is Available

> GCN Team will add notice type to Kafka [Quick Start](../quickstart)

Reviewer: This should be a complete sentence with appropriate articles.

> and Email Notification configuration. Please work with GCN Team on drafting

Reviewer: Add contact link again...

---
handle:
  breadcrumb: New Producers
---

import {
  ProcessList,
  ProcessListItem,
  ProcessListHeading,
  SummaryBox,
} from '@trussworks/react-uswds'

> import { GitHubLabel } from '~/components/GitHubLabel'

Reviewer: These improts are unused. If you don't want to use the process list, then please remove these imports and change the file extension to `.md`.

Author: I actually changed it to use the process list.  Please take a look.

# Setting up New GCN Notice Producers

The following steps guide new instrument, mission, or observatory producers
into setting up new notices streams that are distributed to the user community
via [Kafka](faq#what-is-kafka). This process requires interactions
with the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
to enable these accounts and topics are setup on the GCN Kafka broker,
and we are happy to work together to help you define schema utilizing our
core JSON schema.

## Accounts and Scope

Make sure that all of the producer users (who will enable GCN notices in your pipelines)
have signed in (at least once) to
https://test.gcn.nasa.gov and https://gcn.nasa.gov.

Please send the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
a list of email addresses of all the people who should have the

> producer access to the MissionName.\* topics.

Reviewer: Can these go into the process list?

## Defining Kafka Topics and JSON Schema

The preferred format for formatting of the data sent via GCN Kafka is JSON.

> We have built a set of core schema which serves as the building blocks for

Reviewer 2: Shall we link core-schema Github link?

your Notices. You can also define instrument-specific quantities in your
schema, but we request that you utilize the core schema as much as possible.
Please add your schema to the [GCN repository](https://github.com/nasa-gcn/gcn-schema).

The GCN team is happy to iterate with the producers on their schema contents and format.
Your pipeline will generate JSON files following these schema and send to GCN
in the following process.

## Steps to Start Publishing Alerts

1. Log out and log back in.
2. Go through the "Start Streaming GCN Notices" process.
3. On Step 2 of "Start Streaming GCN Notices", choose the scope gcn.nasa.gov/kafka-MissionName-producer.
4. Your producer code will look very similar to the client example code.
   Start from this and adjust as you see fit:

```sh
from gcn_kafka import Producer

> # Connect as a producer. <br>

Reviewer 3: ```suggestion
# Connect as a producer.
```

# Warning: don't share the client secret with others.

> producer = Producer(client_id='fill me in', _secret='fill me in')

Reviewer 3: ```suggestion
producer = Producer(client_id='fill me in', client_secret='fill me in')
```

topic = 'MissionName.example'
# anything starting with 'MissionName'
data = b'...'  # any bytes
producer.produce(topic, data)
```

## Mission page

Add mission page linking to relevant information and archives, and description of

> new GCN notices content. GCN Team is also available to help construct and format

Reviewer: "You can create a new mission page by submitting a pull request (add link) or by emailing text to the GCN Team (insert link)."

the mission page.

## Announce New Notices Type is Available

GCN Team will add notice type to Kafka [Quick Start](../quickstart)
and Email Notification configuration. Please work with GCN Team on drafting

> community announcement.

Reviewer: ```suggestion
a community announcement.
```

---
handle:
  breadcrumb: New Producers
---

import {
  ProcessList,
  ProcessListItem,
  ProcessListHeading,
} from '@trussworks/react-uswds'

# Setting up New GCN Notice Producers

The following steps guide new instrument, mission, or observatory producers
into setting up new notices streams that are distributed to the user community
via [Kafka](faq#what-is-kafka). This process requires interaction
with the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
to enable accounts and kafka topics creation on the GCN Kafka broker. The GCN Team is
also happy to work with the mission teams to help construct your alerts.

## Steps to Start Publishing Alerts

<ProcessList>
   <ProcessListItem>
     <ProcessListHeading type="h3">

>        Setup Team Member Accounts

Reviewer: ```suggestion
       Set Up Team Member Accounts
```

     </ProcessListHeading>
   - Make sure that your team member(s) who will implement GCN notices in your pipeline

>    has signed in (at least once) to https://test.gcn.nasa.gov and https://gcn.nasa.gov.

Reviewer: ```suggestion
   has signed in at least once to https://test.gcn.nasa.gov and https://gcn.nasa.gov.
```

   - Send the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
   a list of email address(es) of all the team members who should have the

>    producer access to the <code><i>mission_name</i>.\*</code> topics.

Reviewer: Explain where these topics come from. There was probably a missing step here that the user should decide what to call their topics, and that they will be given producer access to all topics beginning with a dotted prefix.

   </ProcessListItem>
   <ProcessListItem>
      <ProcessListHeading type="h3">
         Define Kafka Topics
      </ProcessListHeading>
      Provide your instrument specific kafka topics or topics to the 
      [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn).

>    </ProcessListItem>

Reviewer: What does it mean to "provide a Kafka topic" to us?

   <ProcessListItem>
      <ProcessListHeading type="h3">
         Build Producer Code
      </ProcessListHeading>
      - Log out and log back in.
      - Go through the [Start Streaming GCN Notices](/quickstart) process.
      - On Step 2, choose the scope <code>gcn.nasa.gov/kafka-<i>MissionName</i>-producer</code>.
      - Your producer code will look very similar to the [client example code](docs/client) and
      Step 4 of [Start Streaming GCN Notices](/quickstart). `client_id` and `client_secret`
      can be found in Step 4 client example code.
      - Start from this and adjust the `client_id`, `client_secret`, `topic` and `data` content:

      ```python
      from gcn_kafka import Producer
      # Connect as a producer.
      # Warning: don't share the client secret with others.
      producer = Producer(client_id='fill me in', client_secret='fill me in')
      topic = 'mission_name.example'
      # any topic starting with 'mission_name'
      data = b'...'  # any bytes
      producer.produce(topic, data)
      ```

> 

Reviewer: ```suggestion
```

   </ProcessListItem>
   <ProcessListItem>
      <ProcessListHeading type="h3">
         Create or Update the Mission Page
      </ProcessListHeading>
      Create a new mission page by submitting a [pull request](https://github.com/nasa-gcn/gcn.nasa.gov/pulls)
or by emailing text to the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn).
   </ProcessListItem>

   <ProcessListItem>
      <ProcessListHeading type="h3">
         Announce New Notice Types are Available
      </ProcessListHeading>

> 

Reviewer: ```suggestion
```

      The GCN Team will add your new notice type to the Kafka [Quick Start](../quickstart)
      and Email Notification configuration. Please work with the
      [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
      on drafting a community announcement.

> 

Reviewer: ```suggestion
```

---
handle:
  breadcrumb: New Producers
---

import {
  ProcessList,
  ProcessListItem,
  ProcessListHeading,

>   SummaryBox,

Reviewer: Remove unused import:

```suggestion
```

} from '@trussworks/react-uswds'

> import { GitHubLabel } from '~/components/GitHubLabel'

Reviewer: Remove unused import:

```suggestion
```

# Setting up New GCN Notice Producers

The following steps guide new instrument, mission, or observatory producers
into setting up new notices streams that are distributed to the user community
via [Kafka](faq#what-is-kafka). This process requires interactions
with the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
to enable these accounts and topics are setup on the GCN Kafka broker,
and we are happy to work together to help you define schema utilizing our

> core JSON schema.

Reviewer: This sentence is long and confusing. It has too many long clauses joined by "and", some of which are independent clauses and some of which are dependent classes. Split it into two or more sentences.

## Steps to Start Publishing Alerts

<ProcessList>
   <ProcessListItem>
     <ProcessListHeading type="h3">

>        Team Member Accounts

Reviewer: All of the process list headings should be complete sentences, minus the terminal punctuation.

     </ProcessListHeading>
   - Make sure that your team member(s) who will implement GCN notices in your pipeline
   has signed in (at least once) to https://test.gcn.nasa.gov and https://gcn.nasa.gov.
   - Send the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
   a list of email address(es) of all the team members who should have the
   producer access to the <code><i>mission_name</i>.\*</code> topics.
   </ProcessListItem>
   <ProcessListItem>
      <ProcessListHeading type="h3">
         Defining Kafka Topics and JSON Schema
      </ProcessListHeading>
      The preferred format for formatting of the data sent via GCN Kafka is JSON.

>       We have designed a set of [core schema](https://github.com/nasa-gcn/gcn-schema/tree/master/core-schema)

Reviewer: I don't think that the unified schema is ready for prime time yet. We haven't even checked that they are valid JSON Schema.

Author: validation is mostly addressed.  I removed this, but we do need to get the schema ready to go.

      which serves as the building blocks for your Notices.
      You can also define instrument-specific quantities in your
      schema, but we request that you utilize the core schema as much as possible.
      Please add your schema to the [GCN repository](https://github.com/nasa-gcn/gcn-schema).

      The GCN team is happy to iterate with the producers on their schema contents and format.
      Your pipeline will generate JSON files following these schema and send to GCN
      in the following process.

   </ProcessListItem>
   <ProcessListItem>
      <ProcessListHeading type="h3">
         Producer Example Code
      </ProcessListHeading>
      - Log out and log back in.

>       - Go through the "Start Streaming GCN Notices" process.

Reviewer: ```suggestion
      - Go through the [Start Streaming GCN Notices](/quickstart) process.
```

>       - On Step 2 of ["Start Streaming GCN Notices"](../quickstart),

Reviewer: ```suggestion
      - On Step 2 of [Start Streaming GCN Notices](/quickstart),
```

>       choose the `scope gcn.nasa.gov/kafka-MissionName-producer`.

Reviewer: ```suggestion
      choose the scope <code>gcn.nasa.gov/kafka-<i>MissionName</i>-producer</code>.
```

      - Your producer code will look very similar to the [client example code](docs/client) and

>       Step 4 of ["Start Streaming GCN Notices"](../quickstart). `client_id` and `client_secret`

Reviewer: ```suggestion
      Step 4 of [Start Streaming GCN Notices](/quickstart). `client_id` and `client_secret`
```

      can be found in Step 4 client example code.

>       - Start from this and adjust as you see fit:

Reviewer: Explain what "how you see fit" entails.

      ```python
      from gcn_kafka import Producer
      # Connect as a producer.
      # Warning: don't share the client secret with others.
      producer = Producer(client_id='fill me in', client_secret='fill me in')
      topic = 'mission_name.example'
      # any topic starting with 'mission_name'
      data = b'...'  # any bytes
      producer.produce(topic, data)
      ```

   </ProcessListItem>
   <ProcessListItem>
      <ProcessListHeading type="h3">
         Mission Page
      </ProcessListHeading>

>       Please create a new mission page by submitting a [pull request](https://github.com/nasa-gcn/gcn.nasa.gov/pulls)

Reviewer: Remove the "please." This is a set of instructions.

## Steps to Start Publishing Alerts

1. Make sure that all of the producer users (who will enable GCN notices in your pipelines)
   have signed in (at least once) to https://test.gcn.nasa.gov and https://gcn.nasa.gov.

> 2. Please send the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)

Reviewer: ```suggestion
2. Send the [GCN Team](https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn)
```

   a list of email addresses of all the people who should have the

>    producer access to the MissionName.\* topics.

Reviewer: Add a step: decide on a topic name or a prefix for several topic names.

Also, use `mission_name.*` since we use lower-case topic names.

3. Log out and log back in.
4. Go through the "Start Streaming GCN Notices" process.
5. On Step 2 of "Start Streaming GCN Notices", choose the scope gcn.nasa.gov/kafka-MissionName-producer.
6. Your producer code will look very similar to the client example code.
   Start from this and adjust as you see fit:

> ```sh

Reviewer: ```suggestion
```python
```
